### DISCUSSION TITLE
Is that possible to use remote mariadb?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [appwrite/appwrite#1461] [Q&A] [answered]
Since appwrite is not providing any backup mechanism in the box for db and need to do with recomended tools when in production. Is that possible to use separate database instance of mariadb from remote server to configure on appwrite to use it? This will also provide us more flexibility to backup data as well as create master slaves for scalability.

**Comment by eldadfux** [comment, marked as answer]
Yes it is possible, and Appwrite was designed with something like that in mind. You can connect your own remote MariadDB instance by changing this variables: https://appwrite.io/docs/environment-variables#mariadb

You will also need to init the default set of tables Appwrite needs on your database, you can find them here: https://github.com/appwrite/appwrite/blob/master/app/db/SQL/all.sql

**Reply by AndrewBucklin** [reply]
I'm trying to do the same. Doesn't look like the tables are automatically initialized. The all.sql link you provided leads to a 404. Can you provide the updated location it? Thanks!

_EDIT:  I ended up deploying with the included MariaDB, dumped that database, and imported it into our MySQL cluster. Then I re-deployed and updated the environment variables to point to our MySQL cluster._

**Comment by markosole** [comment]
Hi @eldadfux thnaks for the links and details. This is a bit old thread and I am looking into exact same thing. There are probably a ton of changes and link to all.sql is broken. 

Is there any updated procedure and setup that I can follow in order to connect on remote MariaDB? 

 - I have setup MariaDB 10.x and can connect remotely from same machne where containers are running. 
 - I've also modified .env 

Looks like that table structure as my console (docker-compose up) is throwing error "table ..... x y ... not found". It is expecting structure. 
Interesting is that none of this is available in Docs.

**Reply by eldadfux** [reply]
In newer versions, the main Appwrite container checks for the structure on server startup and creates it if needed.

**Reply by markosole** [reply]
This wasn't the case for me and appwrite 1.3.7. I had to export structure from other table and instance.
